### ISSUE TITLE
Dump file are empty

### ISSUE THREAD

**Issue by zayne-anyra** [d0ublew/dump-index-android-bundle#1]
Thank you for this project. I been working on this app quite sometimes, but still unable to figure out it.

#### I try to run `dump-on-server.js` without make any changes:

`frida -U -f <package_name> -l dump-on-server.js`

Output:

```console
Error: libreactnativejni.so: unable to find export '_ZN8facebook5react22isHermesBytecodeBundleERKNS0_12BundleHeaderE'
    at value (frida/runtime/core.js:250)
    at get_addr (D:\frida\script2.js:119)
    at hookLibrnjni (D:\frida\script2.js:41)
    at onLeave (D:\frida\script2.js:109)
```

File **/storage/emulated/0/Download/index.android.bundle** are empty. Maybe because _**_ZN8facebook5react22isHermesBytecodeBundleERKNS0_12BundleHeaderE**_ do not exist, so I changed it to **__ZN8facebook5react16NativeToJsBridge20getJavaScriptContextEv_**.

Output:

```console
[*] Installing hook for libandroid.so @ 0x75e653447000
[*] Installing hook for libreactnativejni.so @ 0x75e5d64a0000
[*] isHermesBytecodeBundle called
[*] bundle size: 0
[*] Writing bundle to /sdcard/Download/index.android.bundle_1734393595030
[*] Successfully wrote bundle to /sdcard/Download/index.android.bundle_1734393595030
[*] AAssetManager_open index.android.bundle
[*] AAssetManager_open js-modules/UNBUNDLE
```

Same, the "/sdcard/Download/index.android.bundle_1734393595030" are empty.

---

If you don't mind, please take a look at this app: https://file.io/8BgbAUpBvEk6

**Comment by d0ublew** [comment]
Hi @zayne-anyra, seems like the app you are dealing with does not encrypt its `index.android.bundle` file.

```
$ file my-digi-14-0-8/assets/index.android.bundle
my-digi-14-0-8/assets/index.android.bundle: Hermes JavaScript bytecode, version 84
```

Hence, you could directly analyze the file without dumping it.

**Comment by zayne-anyra** [comment]
I though Hermes bytecode was kind of encryption/obfuscation. it kinda confuse, I already disassemble it using [hbctool](https://github.com/Kirlif/HBC-Tool), but it quite hard to get it done.

is it possible to dump the hermes .bundle than see what method or function been use?

**Comment by d0ublew** [comment]
Perhaps you could take a look at this tool [hermes-dec](https://github.com/P1sec/hermes-dec/). Although it could decompile the Hermes bytecode back into JavaScript code, you still need to analyze the code pattern as the control flow is not apparent.

**Comment by zayne-anyra** [comment]
yeah its been a while I'm working on this project but still got no progress 🙃.  ie, in that bundle file contains "encrypt" function, how can I hook at this function? is it possible?
